### ISSUE TITLE
No autocompletion for attributes in comprehension expressions.

### ISSUE THREAD

**Issue by laundmo** [microsoft/pylance-release#2836]
<h3> Environment data </h3>

-   Language Server version: 2022.5.2
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>

When editing the output expression of a list comprehension, Pylance does not provide any intellisense/autocompletion for dot attributes.

<h3> Actual behaviour </h3>

When editing the output expression in a list comprehension, provide the same autocompletion which would be provided for the same type in any other context.

<h3> Logs </h3>

Python Language Server Log (when requesting autocomplete suggestions)
```
(11236) Background analysis message: resumeAnalysis
(11236) Background analysis message: getDiagnosticsForRange
(11236) Background analysis message: getDiagnosticsForRange
(11236) [FG] completion at g:\user\laurin\restructured-imports\t.py:1:7 ...
(11236) [FG]   parsing: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi [fs read 0ms] (29ms)
(11236) [FG]   binding: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi (1ms)
(11236) [FG]   parsing: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\typing.pyi [fs read 0ms] (34ms)
(11236) [FG]   binding: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\typing.pyi (4ms)
(11236) [FG]   parsing: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 0ms] (55ms)
(11236) [FG]   binding: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (1ms)
(11236) [FG]   parsing: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\abc.pyi [fs read 0ms] (1ms)
(11236) [FG]   binding: c:\Users\laurin\.vscode\extensions\ms-python.vscode-pylance-2022.5.2\dist\typeshed-fallback\stdlib\abc.pyi (0ms)
(11236) [FG] completion at g:\user\laurin\restructured-imports\t.py:1:7 [found 0 items] (146ms)

```

<h3> Code Snippet / Additional information </h3>

https://user-images.githubusercontent.com/24855949/169283344-56645eeb-b4d6-4d93-9fb4-8433290f7574.mp4

```python
my_list = [1, 2]
b = [a. for a in my_list]
```

**Comment by erictraut** [comment]
Thanks for the bug report. I've updated the parser to perform better recovery in this case. This results in a more complete parse tree, which allows the type evaluator to determine the type of `a`, which allows the completion provider to suggest appropriate member names. This will be addressed in the next release.

**Comment by laundmo** [comment]
thanks for the fast fix, looking forward to using it!

**Comment by debonte** [comment]
This issue has been fixed in version 2022.5.3, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202253-25-May-2022)
